types/key_value.pp

```diff
@@ -0,0 +1,2 @@
+# An LDAP attribute in the form "key: value"
```

Comment:
I don't think we should create types that have such simple matching. The `Openldap::Attributes` makes sense to me though the `key_value` and `value` seem too simple for their own types. By creating types for these it is just more abstraction and testing than value.

Reply:
Yes, it is a way for me to abstract the complex types in more readable chunks.  The fact that the module only manage strings is adding confusion, I hope that one day an `Openldap::Value` can be more than `String[1]`, in fact at least something like `Variant[Boolean,Integer,String[1]]`, but for now the custom types and provider are basically without tests and do poor error checking so I advance by baby steps.

:100: for adding tests for the types, this will allow to "document" what is not supported and bring support for it in future baby steps!

Reply:
While writing this I am now changing my mind.  It looks like we should only manipulate Strings (typed languages clients for LDAP get value as either a String or a binary data that is then passed to a decoder but this is the responsibility of the consumer. wee cannot do this abstraction wisely).  Will refactor accordingly and add tests.